public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -117,6 +121,26 @@ export class GraphiteDatasource {
     return tags;
   }
 
+  getExploreState(queries: GraphiteQuery[]): Partial<ExploreUrlState> {
```

Comment:
I think we need to change this function to not be so Explore specific or have to use explore types (that are not exposed to plugins as they are internal only). 

This should not need to know about ExploreUrlState, think this function should only be:
interpolateVariablesInQueries(queries: DataQuery[]): DataQuery[]  

Will make it easier for data sources to implement and they do not need to use explore specific types.

Reply:
I agree. So does this mean that we should create `interpolateVariablesInQueries` function that needs to be implemented in every datasource? Also then for plugin datasources, do we want to add in documentation, that they should implement `interpolateVariablesInQueries` function?

Reply:
it needs to be optional